add dockerfile and gha to generate docker images

brianhenryhf opened the pull request:
https://www.pivotaltracker.com/n/projects/2663780/stories/188414342

Provides Dockerfile for GHA-based builds. 

Operates off of config files in normal rails directory structure for k8s - with either ENV vars or, where absolutely necessary, a file override in k8s deploy. 

For ec2/capistrano, the following files will continue to be overridden: `'config/mongoid.yml', 'config/initializers/devise.rb', 'config/secrets.yml', "config/environments/production.rb", "config/symmetric-encryption.yml",'config/saml.yml', 'config/unicorn.rb', 'eyes/enroll.eye.rb', 'config/ssl/wfpk.pem'`

polographer commented on bearer.ignore:
is it possible to scope the ignores to a specific file ? like, only ignore the ssn on this particular file ? maybe the hash implicitly does this, so maybe I just need reassurance this are scoped

brianhenryhf replied:
The docs could be better on this, but 2 bits of data from poking around:
1- from some experimentation, adding a new violation of the same type (for example, https://docs.bearer.com/reference/rules/ruby_lang_file_generation/) in another file (or another line of same file) will fail a subsequent bearer check. Adding a new violation of another type to the same file also fails a subsequent bearer check.
2- From looking at the code, this is likely where the fingerprints get generated: https://github.com/Bearer/bearer/blob/main/pkg/report/output/security/security.go#L210-L248 - It looks like it's fingerprinting specifically based on filename and count of occurrences of a given rule in that file.

So, we should be safe there. But, another option that might be clearer would be to use this feature, putting the ignore in a code comment:
https://docs.bearer.com/guides/configure-scan/#:~:text=%2C%20ruby_lang_cookies%5D-,Skip%20rules%20for%20individual%20code%20blocks,-Bearer%20CLI%20supports

brianhenryhf commented:
Rebased against master to clear up some conflict reports in the PR. 
Also, removed `bearer.ignore`, as it's not needed with the adjusted merge target (and the context-less nature of that file seems not ideal w/r/t security errors - whitelisting code without obvious way to see what is being whitelisted. If I were re-adding exceptions, I'd use the code comment mechanism mentioned [here](https://github.com/health-connector/enroll/pull/2818#discussion_r1813324328))

polographer commented on Gemfile:
```diff
@@ -97,6 +97,7 @@ gem 'sassc',                    '~> 2.0'
 gem 'sass-rails',               '~> 5'
```
I think this line got dup in the merge, I don't remember what will happen, maybe bundler will just ignore it
